src/bandersnatch/package.py

```diff
@@ -142,17 +141,13 @@ async def sync(self, stop_on_error: bool = False, attempts: int = 3) -> None:
             while self.tries < attempts:
                 try:
                     logger.info(f"Syncing package: {self.name} (serial {self.serial})")
-                    try:
-                        metadata_generator = self.mirror.master.get(
-                            f"/pypi/{self.name}/json", int(self.serial)
-                        )
-                        metadata_response = await metadata_generator.asend(None)
-                        metadata = await metadata_response.json()
-                    except ClientResponseError as e:
-                        if e.status == 404:
-                            logger.info(f"{self.name} no longer exists on PyPI")
-                            return None
-                        raise
+
+                    metadata = await self.mirror.master.get_package_metadata(
+                        self.name, serial=self.serial
+                    )
+                    if not metadata:
+                        return None
```

Comment:
Would a custom error type be better?  e.g. have the ```get_package_metadata``` function ```raise PackageNotFound(f"{self.name} no longer exists on PyPI"``` and then catch and log and return as before?